### ISSUE TITLE
Attribute aliases

### ISSUE THREAD

**Issue by pyromaniac** [dry-rb/dry-schema#329]
Hey folks! What do you think of attribute aliases?

## Examples

```ruby
schema = Dry::Schema.Params do
  required(:model_id).value(:integer).alias(:id)
end

schema.(model_id: '42').to_h #=> { model_id: 42 }
schema.(id: '42').to_h #=> { model_id: 42 }
schema.(model_id: '42', id: '43').to_h #=> { model_id: 42 }, id is ignored
```

I realize that it can be now done with `before(:key_validator)` but a short-cut would be just perfect.
I'm ready to implement it if you approve the idea.

**Comment by solnic** [comment]
This will be supported once we have dynamic keys feature implemented, which is scheduled for 2.0.0. You'll be able to do things like `required(/model_id|id/).value(:integer)` etc.

**Comment by pyromaniac** [comment]
Thanks! Looking forward to it!
